version: "3.3"

networks:
  rlmp:
    driver: bridge

volumes:
  db:
  cache:
  cache-cfg:

services:
  nginx:
    build: nginx
    container_name: rlmp-nginx
    ports:
      - "8000:80"
    restart: on-failure
    networks:
      - rlmp
    depends_on:
      - api
      - db

  api:
    build: api
    container_name: rlmp-api
    command: go run ./cmd/app/main.go
    restart: on-failure
    ports:
      - "8001:8001"
    environment:
      - RLMP_CONFIGS_FOLDER_PATH=/usr/src/app/configs
      - RLMP_CONFIG_NAME=config
      - RLMP_DOTENV_PATH=/usr/src/app/.env
    depends_on:
      - db
    networks:
      - rlmp

  db:
    container_name: rlmp-db
    image: postgres:latest
    restart: on-failure
    env_file:
      - postgres/db.env
    ports:
      - "5433:5433"
    volumes:
      - db:/var/lib/postgresql/data
    networks:
      - rlmp

  cache:
    container_name: rlmp-cache
    image: redis:6.0.9-alpine
    restart: on-failure
    environment:
      - REDIS_REPLICATION_MODE=master
    volumes:
      - cache:/var/lib/redis
      - cache-cfg:/usr/local/etc/redis/redis.conf
    networks:
      - rlmp

  parser:
    build: universities_directions_parser
    container_name: rlmp-parser
    command: python3 ./app.py
    restart: on-failure
    networks:
      - rlmp
    depends_on:
      - api
      - db
      - cache
      - nginx
